### PR TITLE
[Module method]-[development] implemented module method through JQuery AJAX

### PR DESCRIPTION
**Pull request by kodeerkarthik**

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -3,6 +3,100 @@ user
 continue
```
Comment (Sujina-Saddival) [inline .byebug_history]: why is this needed?

File: app/controllers/manage_user_controller.rb [hunk 2]
```diff
@@ -5,4 +5,10 @@ def index
     @users = User.all
     @articles = Article.all
   end
+
+  def user_roles
```
Comment (Sujina-Saddival) [inline app/controllers/manage_user_controller.rb]: change the method name, You are fetching user

File: app/controllers/permissions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class PermissionsController < ApplicationController
+	include PermissionsHelper
```
Comment (Sujina-Saddival) [inline app/controllers/permissions_controller.rb]: Can you change the controller name?

File: app/helpers/permissions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+module PermissionsHelper
```
Comment (Sujina-Saddival) [inline app/helpers/permissions_helper.rb]: Good work

File: app/models/ability.rb [hunk 5]
```diff
@@ -5,13 +5,21 @@ class Ability
...
+
+      if user.has_role? :admin
         can :manage, :all
       else
-        can :manage, Article, User_id: user.id 
         can :read, Article
-        can :read, Post
-        can :manage, Post, User_id: user.id 
+        can :manage, Article if user.has_role?(:manager, Article)
+        can :write, Article, :id => Article.with_role(:moderator, user).pluck(:id)
```
Comment (Sujina-Saddival) [inline app/models/ability.rb]: after plucking the id's.For what are using it?

File: app/views/articles/show.html.erb [hunk 6]
```diff
@@ -9,10 +9,10 @@
       <strong>Text:</strong>
       <%= @article.text %>
     </p>
-    <% if can? :update, @article %>
+    <% if current_user.has_role? :admin or current_user.has_role? :manager or current_user.has_role? :moderator %>
```
Comment (Sujina-Saddival) [inline app/views/articles/show.html.erb]: write a method like and use these methods where ever u want
current_user.admin?
current_user.manager?
current_user.moderator?

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -20,7 +20,7 @@
     <%= image_tag("logo.png",:width=>'100', :height=>'30' )%>
     <% if user_signed_in? %>
       <%= link_to 'Home', home_path, :class => "btn btn-outline-info"%>
-      <% if can? :manage, :all %>
+      <% if current_user.has_role? :admin or current_user.has_role? :manager %>
```
Comment (Sujina-Saddival) [inline app/views/layouts/application.html.erb]: same thing

File: app/views/manage_user/index.html.erb [hunk 8]
```diff
@@ -2,6 +2,8 @@
 	<body class='index_body' id='index_body'>
 		<div class='shade pt-5'>
 			<div class="accordion" id="accordionExample">
+			<%= link_to 'define role', permissions_path, :class=>'btn btn-danger btn-sm' %>
```
Comment (Sujina-Saddival) [inline app/views/manage_user/index.html.erb]: Why danger class? Dont want red button. Change to some other class

File: app/views/permissions/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,43 @@
+<div class='container mt-5'>
```
Comment (Sujina-Saddival) [inline app/views/permissions/show.html.erb]: try to change the file/controller from permission to USER

File: app/views/permissions/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+	<div class='mb-3'><b>Email : </b><%= @user.email%></div>
+	<div class='mb-3'><b>Username : </b><%= @user.username%></div>
+	<div class='mb-3'><b>Role : </b>
+		<% if @user.roles.first.present? %>
```
Comment (Sujina-Saddival) [inline app/views/permissions/show.html.erb]: change this to @user.roles.any?

File: app/views/permissions/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+			<%= @user.roles.last.name%><br>
+		<%else%>
+			no role assigned<br>
+		<%end%>
+	</div>
+	
+	<%= select_tag(:role, 
+	options_for_select(['admin','moderator','manager','viewer'],
+	if @user.roles.first.present?
```
Comment (Sujina-Saddival) [inline app/views/permissions/show.html.erb]: change this to @user.roles.any?

File: app/views/permissions/show.html.erb [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+	<div class='mb-3'><b>Role : </b>
+		<% if @user.roles.first.present? %>
+			<%= @user.roles.last.name%><br>
+		<%else%>
+			no role assigned<br>
+		<%end%>
+	</div>
+	
+	<%= select_tag(:role, 
+	options_for_select(['admin','moderator','manager','viewer'],
```
Comment (Sujina-Saddival) [inline app/views/permissions/show.html.erb]: define these types of array ['admin','moderator','manager','viewer'] in a separate file. Create a file as constant and define the array/value. import the constant where ever u want

File: app/views/permissions/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+		@user.roles.last.name 
+	end), :id=>'select', :user_id=>@user.id) %>	
+	<%= button_tag "remove role", :id=>"rmrole", :user_id=>@user.id, :class=>'btn btn-danger btn-sm'%><br>
+	<%= link_to 'Back', permissions_path, :class=>"btn btn-dark btn-sm mt-4" %>
+</div>
+
+
+<script>
+	$(document).ready(function(){
+		$("#select").change(function(){
```
Comment (Sujina-Saddival) [inline app/views/permissions/show.html.erb]: Good work, But it/class name be more specific

File: db/schema.rb [hunk 14]
```diff
@@ -11,7 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
```
Comment (Sujina-Saddival) [inline db/schema.rb]: **PLEASE DO NOT PUSH SCHEMA.rb** Before raising please you review once

File: app/views/articles/show.html.erb [hunk 15]
```diff
@@ -9,10 +10,10 @@
       <strong>Text:</strong>
       <%= @article.text %>
     </p>
-    <% if can? :update, @article %>
+    <% if current_user.admin or current_user.moderator or current_user.manager%>
```
Comment (Sujina-Saddival) [inline app/views/articles/show.html.erb:13]: user || operation

File: db/schema.rb (hunk 14 above)
Comment (Sujina-Saddival) [reply db/schema.rb]: ?

File: app/models/ability.rb (hunk 5 above)
Comment (Sujina-Saddival) [reply app/models/ability.rb]: ?

File: .byebug_history (hunk 1 above)
Comment (Sujina-Saddival) [reply .byebug_history]: ?

Comment (kodeerkarthik) [reply .byebug_history]: This one was automatically generated. I think it keeps track of debugger history.

File: app/controllers/manage_user_controller.rb (hunk 2 above)
Comment (kodeerkarthik) [reply app/controllers/manage_user_controller.rb]: I removed this method (unwanted method)

File: app/controllers/permissions_controller.rb (hunk 3 above)
Comment (kodeerkarthik) [reply app/controllers/permissions_controller.rb]: replaced this Permissionscontroller with the existing one, which is the manage_users controller.

File: app/helpers/permissions_helper.rb (hunk 4 above)
Comment (kodeerkarthik) [reply app/helpers/permissions_helper.rb]: Thank you

File: app/models/ability.rb (hunk 5 above)
Comment (kodeerkarthik) [reply app/models/ability.rb]: I removed this for unwanted code.

File: app/views/articles/show.html.erb (hunk 6 above)
Comment (kodeerkarthik) [reply app/views/articles/show.html.erb]: have created methods as you mentioned above, and use those methods by invoking using their objects (class variables)

File: app/views/layouts/application.html.erb (hunk 7 above)
Comment (kodeerkarthik) [reply app/views/layouts/application.html.erb]: resolved.

File: app/views/manage_user/index.html.erb (hunk 8 above)
Comment (kodeerkarthik) [reply app/views/manage_user/index.html.erb]: removed the link because I replaced the controller from the new controller to the existing one.

File: app/views/permissions/show.html.erb (hunk 9 above)
Comment (kodeerkarthik) [reply app/views/permissions/show.html.erb]: replaced the controller with the existing one

File: app/views/permissions/show.html.erb (hunk 10 above)
Comment (kodeerkarthik) [reply app/views/permissions/show.html.erb]: completed

File: app/views/permissions/show.html.erb (hunk 11 above)
Comment (kodeerkarthik) [reply app/views/permissions/show.html.erb]: completed

File: app/views/permissions/show.html.erb (hunk 12 above)
Comment (kodeerkarthik) [reply app/views/permissions/show.html.erb]: completed

File: app/views/permissions/show.html.erb (hunk 13 above)
Comment (kodeerkarthik) [reply app/views/permissions/show.html.erb]: done

File: db/schema.rb (hunk 14 above)
Comment (kodeerkarthik) [reply db/schema.rb]: I have added this file to git ignore file but I don't know, how it was pushing into the repo.
next time I will take care of it.
